e2e/test/scenarios/sharing/downloads/downloads.cy.spec.js

```diff
@@ -185,47 +190,37 @@ describe("scenarios > question > download", () => {
       { visitQuestion: true },
     );
 
+    cy.findByRole("button", { name: "Download results" }).click();
+    H.popover().within(() => {
+      cy.findByText(".xlsx")
+        .should("be.visible")
+        .click()
+        .then(() => {
+          cy.wait("@saveFormat");
+        });
+    });
+
     cy.intercept(
       "GET",
       "/api/user-key-value/namespace/last_download_format/key/download_format_preference",
     ).as("fetchFormat");
 
+    cy.reload().then(() => {
+      cy.wait("@fetchFormat").then(() => {
+        cy.findByRole("button", { name: "Download results" }).click();
+        H.popover().within(() => {
+          cy.get("[data-checked='true']").should("contain", ".xlsx");
+        });
+      });
+    });
+  });
```

Comment:
I don't think these nested `.then()`s should change anything about the behavior. By default cypress does all of these things synchronously